modules/rubiconBidAdapter.js

```diff
@@ -212,6 +214,14 @@ export const converter = ortbConverter({
 
     setBidFloors(bidRequest, imp);
 
+    // ensure unique imp IDs for twin adunits
+    let adUnitCode = imp.id;
+    let i = 2;
+    while (Object.keys(impIdMap).indexOf(imp.id) > -1) {
+      imp.id = adUnitCode + i++;
+    }
+    impIdMap[imp.id] = adUnitCode;
```

Comment:
I don't think a loop is needed here

```js
if (impIdMap[imp.id]) {
   imp.id = imp.id + impIdMap[imp.id];
   impIdMap[imp.id] += 1
} else {
   impIdMap[imp.id] = 2;
}
```

The above can probably even be a bit simplified probably

But the gist is

- Make a object

{
   [name of adunit]: [how many times its been used]
}

Then no looping is needed and if the adunit exists in the map, just append the value to the id and increment.

Let me know if that works / makes sense!

Reply:
All good, I've made the change :-)